### PR TITLE
Extract instantiation of cloud provider

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

Add a separate method in a new file for creating cloud providers.
Currently the code is all mixed into the controller manager. We
should actively control what is made available to the cloud provider
so list explicitly the parms needed and move the code out. This will
avoid linkages to sneak in as we will catch it better during reviews.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
context : please see https://github.com/kubernetes/kubernetes/pull/59440

File: cmd/kube-controller-manager/app/cloudproviders.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"k8s.io/client-go/informers"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+// createCloudProvider helps consolidate what is needed for cloud providers, we explicitly list the things
+// that the cloud providers need as parameters, so we can control
+func createCloudProvider(cloudProvider string, externalCloudVolumePlugin string, cloudConfigFile string,
+	allowUntaggedCloud bool, sharedInformers informers.SharedInformerFactory) (cloudprovider.Interface, ControllerLoopMode, error) {
+	var cloud cloudprovider.Interface
+	var loopMode ControllerLoopMode
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/cloudproviders.go]: huh, didn't see this coming.  I'm really glad this got extracted.

File: cmd/kube-controller-manager/app/cloudproviders.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	var err error
+	if cloudprovider.IsExternal(cloudProvider) {
+		loopMode = ExternalLoops
+		if externalCloudVolumePlugin != "" {
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/cloudproviders.go]: nit: invert this check and return early.

**Comment by deads2k** [comment]
@dims I'm surprised how deeply got it's hooks in. Thanks for pulling it out.

I left a nit if you get to in time, otherwise it can be cleaned up some other time.

/lgtm

**Pull request merged**